v0/destinations/singular/transform.js

```diff
@@ -0,0 +1,215 @@
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_URL,
+  sessionEvents,
+  SINGULAR_SESSION_ANDROID_EXCLUSION,
+  SINGULAR_SESSION_IOS_EXCLUSION,
+  SINGULAR_EVENT_ANDROID_EXCLUSION,
+  SINGULAR_EVENT_IOS_EXCLUSION
+} = require("./config");
+const {
+  defaultRequestConfig,
+  constructPayload,
+  CustomError,
+  defaultGetRequestConfig,
+  removeUndefinedAndNullAndEmptyValues,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  extractCustomFields,
+  getValueFromMessage
+} = require("../../util");
+
+const responseBuilderSimple = (message, { Config }) => {
+  // trim and replace spaces with "_"
+  message.event = message.event.trim().replace(/\s+/g, " ");
+  const eventType = message.event;
+  const response = defaultRequestConfig();
+  const eventAttributes = {};
+  let payload, endPoint;
+  const platform = getValueFromMessage(message, "context.os.name");
```

Comment:
We should put a check if `platform` is defined or not.

Reply:
Done